### ISSUE TITLE
"POD.exit" should be allocated only if needed.

### ISSUE THREAD

**Issue by pierrethibault** [ptycho/ptypy#43]
To make POD's behaviour uniform, a "fake" exit view `self._exit` is created in `POD.__init__`. This is a waste of memory. Possible solution: create the array on-the-fly when `POD.exit` is accessed, or store the array as a class variable (which would have to depend on the geometry).

**Comment by pierrethibault** [comment]
Not as simple: `self._exit` can be set to something else than `ones`. Why are there two ways to store exit waves?

**Comment by bjoernenders** [comment]
At some point the container access from the exit views was slowing the code down. I never w quite understood why, but having an exit view for each pod resolved this problem. At later stages of development this bug resolved itself when the layer access was recoded. That is why the exit-wave Container was resurrected. However I left the private exit wave as a switch option in case a mysterious exit-wave slow down reappears. The default behavior now is to use a container which is the canonic way of the package. If there are now more complains regarding speed for larger datasets, we may ditch the option of a private exit wave in future releases. Thanks for pinpointing to this inconsistency

**Comment by pierrethibault** [comment]
Ok. Still shouldn't we consider the possibility of _not_ initialising the arrays on POD construction because some algorithms such as ML (and ePIE) do not need this.

**Comment by alexbjorling** [comment]
Right, for ePIE you actually don't need to allocate memory for the exit waves at all, you just need one exit wave buffer per node that you can reuse for all its PODs. So (if I have this straight) the memory requirement for ePIE is essentially half that of DM. Especially since ePIE seems to have much worse convergence than DM to start with, we should exploit this advantage.

**Comment by bjoernenders** [comment]
We have this in parts now, though I think we should review the practice in the package. I think the 0.9 release will be the right target for this review.
